pkg/sources/source_unit.go

```diff
@@ -1,5 +1,13 @@
 package sources
 
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Ensure CommonSourceUnit implements SourceUnit at compile time.
```

Comment:
nit: `// SourceUnitID implements the SourceUnit interface.`
Just to be pedantic and follow the go style guide around exported methods 🤣 
https://google.github.io/styleguide/go/decisions#doc-comments